### PR TITLE
maintainers: add ofek

### PR DESCRIPTION
**Pull request by ofek**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/pull/163088#issuecomment-1060535001

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/hatchling/default.nix
Comment (Artturin) [inline pkgs/development/python-modules/hatchling/default.nix:10]:
```suggestion
, editables
```
`? null` should not be used in nixpkgs

File: pkgs/development/python-modules/hatchling/default.nix
Comment (Artturin) [inline pkgs/development/python-modules/hatchling/default.nix:30]: https://github.com/takluyver/intreehooks readme says this is obsolete?

File: pkgs/development/python-modules/hatchling/default.nix
Comment (Artturin) [inline pkgs/development/python-modules/hatchling/default.nix:46]: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies-handling-dependencies
no need to have python deps in buildInputs unless they're build time only

File: pkgs/development/python-modules/hatchling/default.nix
Comment (Artturin) [inline pkgs/development/python-modules/hatchling/default.nix:1]: no need to have this

**Review by Artturin — CHANGES_REQUESTED** [review]
Please read
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
and 
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-contributing

**Comment by ofek** [comment]
Thanks, fixed!

File: pkgs/development/python-modules/hatchling/default.nix
Comment (Artturin) [inline pkgs/development/python-modules/hatchling/default.nix:33]: forgot to put this into my previous review but we would like to remove/minimize python2 packages from nixpkgs so please no new python2 packages
https://github.com/NixOS/nixpkgs/projects/27

**Comment by Artturin** [comment]
the `default.nix` file should be called from `pkgs/top-level/python-packages.nix` otherwise it cannot be used

**Comment by ofek** [comment]
Addressed, thank you!

File: pkgs/development/python-modules/hatchling/default.nix
Comment (Artturin) [inline pkgs/development/python-modules/hatchling/default.nix:43]: a entry for `ofek` has to be added to `./maintainers/maintainer-list.nix`
https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers
the `maintainers: add ofek` commit should be before the init commit

File: pkgs/development/python-modules/hatchling/default.nix
Comment (Artturin) [inline pkgs/development/python-modules/hatchling/default.nix:30]: `nix-build . -A python3Packages.hatchling`

build fails with
```
ERROR: Could not find a version that satisfies the requirement tomli~=2.0.0; python_version > "3" (from hatchling) (from versions: none)
ERROR: No matching distribution found for tomli~=2.0.0; python_version > "3"
```

sadly the tomli update pr hasn't been merged and it also will go in to staging which will take some time to reach the master branch https://github.com/NixOS/nixpkgs/pull/157735

i will test the builds and if they succeed then merge


so we can either leave this unmerged and then i merge this once the tomli pr is in master or i merge this in to staging

BTW afaik the hatchling does not need to be in nixpkgs for poetry2nix to be able to use it though since hatchling is not poetry, poetry2nix will need to add support for it

**Comment by ofek** [comment]
I reordered the commits as requested and relaxed the dependency constraints. Can you please try building again?

File: pkgs/development/python-modules/hatchling/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hatchling/default.nix:6]:
```suggestion
```

File: pkgs/development/python-modules/hatchling/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/hatchling/default.nix:37]: Why? The project has tests https://github.com/ofek/hatch/tree/master/tests and missing pythonImportsCheck.

**Comment by mweinelt** [comment]
Already part of python-updates for a datadogpy update.
https://github.com/NixOS/nixpkgs/pull/162538/commits/75e47c35288f416bd46869386ed49ccc72be640e

File: pkgs/development/python-modules/hatchling/default.nix
Comment (mweinelt) [reply pkgs/development/python-modules/hatchling/default.nix:37]: Because the tests try to pull library from the internet.

```
running install tests
<<<<< Copying backend >>>>>
<<<<< Building backend >>>>>
* Creating virtualenv isolated environment...
* Getting dependencies for wheel...
* Installing packages in isolated environment... (editables>=0.2; python_version > "3", importlib-metadata; python_version < "3.8", packaging>=21.3; python_version > "3", packaging~=20.9; python_version < "3", pathspec>=0.9, pluggy>=1.0.0; python_version > "3", pluggy~=0.13; python_version < "3", tomli>=1.2.2; python_version > "3", toml~=0.10.2; python_version < "3")
Ignoring packaging: markers 'python_version < "3"' don't match your environment
Ignoring toml: markers 'python_version < "3"' don't match your environment
Ignoring pluggy: markers 'python_version < "3"' don't match your environment
Ignoring importlib-metadata: markers 'python_version < "3.8"' don't match your environment
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff59be410>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pathspec/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff59bfe80>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pathspec/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff59be0e0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pathspec/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff59be110>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pathspec/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7ffff56fa110>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/pathspec/
ERROR: Could not find a version that satisfies the requirement pathspec>=0.9 (from versions: none)
ERROR: No matching distribution found for pathspec>=0.9
```

Comment (SuperSandro2000) [reply pkgs/development/python-modules/hatchling/default.nix:37]: Ah, it tries to install things into a virtualenv.

**Comment by ofek** [comment]
@mweinelt Oh yes please, thanks! If it's alright with you can we use `fetchPypi` instead of `fetchFromGitHub`? The repo houses 2 projects and I don't want a structure change to disrupt anything.

@Artturin I updated to only add myself to the list of maintainers.

**Comment by mweinelt** [comment]
As described here the tests currently require the Git repository structure. https://github.com/ofek/hatch/issues/120#issuecomment-1030840491

If that gets fixed and we can run some tests offline that would certainly be appealing.

**Comment by ofek** [comment]
> tests currently require the Git repository structure

Fixed:

- https://github.com/ofek/hatch/commit/165fc1fbabef071508db81eb84fce28eebe44873
- https://pypi.org/project/hatchling/0.20.1/

**Review by jonringer — APPROVED** [review]

**Pull request merged**
